HttpFs/HttpFs.fs

```diff
@@ -523,195 +528,292 @@ module Client =
 
     open StreamWriters
 
-    let generateFileData (encoding : Encoding) contentType contents = seq {
-      match contentType, contents with
-      | { typ = "text"; subtype = _ }, Plain text ->
-        yield writeLineAscii ""
-        yield writeLineUtf8 text
-
-      | { typ = "application"; subtype = subtype }, Plain text 
-        when List.exists ((=) (subtype.Split('+') |> Seq.last)) ["json"; "xml"] ->
-        yield writeLineAscii ""
-        yield writeLineUtf8 text
-
-      | _, Plain text ->
-        yield writeLineAscii "Content-Transfer-Encoding: base64"
-        yield writeLineAscii ""
-        yield writeLineAscii (text |> encoding.GetBytes |> Convert.ToBase64String)
-
-      | _, Binary bytes ->
-        yield writeLineAscii "Content-Transfer-Encoding: binary"
-        yield writeLineAscii ""
-        yield writeBytesLine bytes
-
-      | _, StreamData stream ->
-        yield writeLineAscii "Content-Transfer-Encoding: binary"
-        yield writeLineAscii ""
-        yield writeStreamLine stream
-    }
+    let createFileHttpContent (encoding : Encoding) contentType (contentDisposition: string) contents : HttpContent =
+      // match contentType, contents with
+      // | { typ = "text"; subtype = _ }, Plain text ->
+      //   yield writeLineAscii ""
+      //   yield writeLineUtf8 text
+
+      // | { typ = "application"; subtype = subtype }, Plain text
+      //   when List.exists ((=) (subtype.Split('+') |> Seq.last)) ["json"; "xml"] ->
+      //   yield writeLineAscii ""
+      //   yield writeLineUtf8 text
+
+      // | _, Plain text ->
+      //   yield writeLineAscii "Content-Transfer-Encoding: base64"
+      //   yield writeLineAscii ""
+      //   yield writeLineAscii (text |> encoding.GetBytes |> Convert.ToBase64String)
+
+      // | _, Binary bytes ->
+      //   yield writeLineAscii "Content-Transfer-Encoding: binary"
+      //   yield writeLineAscii ""
+      //   yield writeBytesLine bytes
+
+      // | _, StreamData stream ->
+      //   yield writeLineAscii "Content-Transfer-Encoding: binary"
+      //   yield writeLineAscii ""
+      //   yield writeStreamLine stream
+      let content =
+        match contentType, contents with
+        | { typ = "text"; subtype = _ }, Plain text ->
+          new StringContent(text, Encoding.UTF8) :> HttpContent
+
+        | { typ = "application"; subtype = subtype }, Plain text
+          when List.exists((=) (subtype.Split('+') |> Seq.last)) ["json"; "xml"] ->
+          new StringContent(text, Encoding.UTF8) :> HttpContent
+
+        | _, Plain text ->
+          let cnt = new StringContent(text |> encoding.GetBytes |> Convert.ToBase64String, Encoding.ASCII)
+          cnt.Headers.Add("Content-Transfer-Encoding", "base64")
+          cnt :> HttpContent
+
+        | _, Binary bytes ->
+          let cnt = new ByteArrayContent(bytes)
```

Comment:
Why use their implementation? I'd prefer it if we can keep a transparent functional implementation, don't you think?

Reply:
I'm not against transparently writing the content. My first attempt actually did that. I used StreamContent and a MemoryStream to write the output content with the original code. It worked, with one problem that Suave started throwing TCP errors on each request:
```
TCP request processing failed
System.Exception: invalid url: ''
   at <StartupCode$Suave>.$ConnectionFacade.processRequest@439-2.Invoke(String _arg29)
   at Microsoft.FSharp.Control.AsyncBuilderImpl.args@823-1.Invoke(a a)
```
Everything seemed to work even with the warning, but the TCP connections were being dropped, so poor performance. I swapped to the other HttpContent types to test and it worked, so stopped my investigation there.

I can give it another try?

Reply:
I will change it back to be similar to the original.
HttpClient is slightly different in that the request stream is not directly exposed like it was with `webRequest.GetRequestStream()`. Without introducing a new HttpContent type, the simplest approach I think is:
```fsharp
use ms = new MemoryStream()
...Write content to memory stream...
requestMessage.Content <- new StreamContent(ms)
```